server/src/main/java/org/apache/druid/server/coordination/ServerManager.java

```diff
@@ -290,6 +286,13 @@ private <T> QueryRunner<T> buildAndDecorateQueryRunner(
     if (segmentId == null || segmentInterval == null) {
       return new ReportTimelineMissingSegmentQueryRunner<>(segmentDescriptor);
     }
+
+    StorageAdapter storageAdapter = segment.asStorageAdapter();
```

Comment:
nit. could just make this final

Reply:
following the same convention as the rest of this code is written in.